odbc/src/api/functions.rs

```diff
@@ -1785,11 +1785,441 @@ pub unsafe extern "C" fn SQLGetInfo(
+        }
+        // SQL_TIMEDATE_ADD_INTERVALS
+        InfoType::TimedateAddIntervals
+        // SQL_TIMEDATE_DIFF_INTERVALS
+        | InfoType::TimedateDiffIntervals => {
+            // MongoSQL DATEADD and DATEDIFF support the following intervals.
+            const TIMEDATE_INTERVALS: u32 = SQL_FN_TSI_SECOND
+                | SQL_FN_TSI_MINUTE
+                | SQL_FN_TSI_HOUR
+                | SQL_FN_TSI_DAY
+                | SQL_FN_TSI_WEEK
+                | SQL_FN_TSI_MONTH
+                | SQL_FN_TSI_QUARTER
+                | SQL_FN_TSI_YEAR;
+            return i16_len::set_output_fixed_data(&TIMEDATE_INTERVALS, info_value_ptr, string_length_ptr);
+        }
+        // SQL_CATALOG_LOCATION
+        InfoType::CatalogLocation => {
+            // MongoSQL puts the catalog (database) at the start of a qualified
+            // table name. As in, db.table.
+            return i16_len::set_output_fixed_data(&SQL_CL_START, info_value_ptr, string_length_ptr);
+        }
+        // SQL_SQL_CONFORMANCE
+        InfoType::SqlConformance => {
+            // MongoSQL is SQL-92 Entry level compliant.
+            return i16_len::set_output_fixed_data(
+                &SQL_SC_SQL92_ENTRY,
+                info_value_ptr,
+                string_length_ptr,
+            );
+        }
+        // SQL_ODBC_INTERFACE_CONFORMANCE
+        InfoType::OdbcInterfaceConformance => {
+            // The MongoSQL ODBC Driver is currently meets the minimum compliance level.
+            return i16_len::set_output_fixed_data(&SQL_OIC_CORE, info_value_ptr, string_length_ptr);
+        }
+        // SQL_SQL92_PREDICATES
+        InfoType::Sql92Predicates => {
+            // MongoSQL supports the following SQL-92 predicate operators.
+            const PREDICATES: u32 = SQL_SP_EXISTS
+                | SQL_SP_ISNOTNULL
+                | SQL_SP_ISNULL
+                | SQL_SP_LIKE
+                | SQL_SP_IN
+                | SQL_SP_BETWEEN
+                | SQL_SP_COMPARISON
+                | SQL_SP_QUANTIFIED_COMPARISON;
+            return i16_len::set_output_fixed_data(&PREDICATES, info_value_ptr, string_length_ptr);
+        }
+        // SQL_SQL92_RELATIONAL_JOIN_OPERATORS
+        InfoType::Sql92RelationalJoinOperators => {
+            // MongoSQL supports the following SQL-92 JOIN operators.
+            const JOIN_OPS: u32 = SQL_SRJO_CROSS_JOIN
+                | SQL_SRJO_INNER_JOIN
+                | SQL_SRJO_LEFT_OUTER_JOIN
+                | SQL_SRJO_RIGHT_OUTER_JOIN;
+            return i16_len::set_output_fixed_data(&JOIN_OPS, info_value_ptr, string_length_ptr);
+        }
+        // SQL_AGGREGATE_FUNCTIONS
+        InfoType::AggregateFunctions => {
+            // MongoSQL supports the following aggregate functions.
+            const AGG_FUNCTIONS: u32 = SQL_AF_AVG
+                | SQL_AF_COUNT
+                | SQL_AF_MAX
+                | SQL_AF_MIN
+                | SQL_AF_SUM
+                | SQL_AF_DISTINCT
```

Comment:
I don't know if we are supposed to return SQL_AF_DISTINCT because we don't support `select Distinct col1, col2 from table1` for example. I couldn't find a description of what SQL_AF_DISTINCT stands for however, so I am not 100% sure.

Reply:
My impression was that `SQL_AF_DISTINCT` (standing for "SQL AGGREGATE FUNCTION DISTINCT") corresponded to the `DISTINCT` keyword being used in aggregate functions -- which we do support. The `SQL_AF_*` values all correspond to aggregation functions (`SUM`, `MAX`, etc.), except for two: `SQL_AF_ALL` and `SQL_AF_DISTINCT`. This is why I assumed they were modifiers for aggregation functions.

I doubt it is a meant to indicate `DISTINCT` as in `SELECT DISTINCT`, but I suppose it is possible! Do you know where I can find documentation on the header file values? The files themselves have no documentation, and the docs at `https://learn.microsoft.com/en-us/sql/odbc/reference/...` don't always specifically spell out exactly what each one means.

Reply:
Yes, this is my problem here too.
I don't know what SQL_AF_DISTINCT stands for and couldn't find the answer. Couldn't find anything useful in the spec and the .h file has no comment :(
I think you can read it both ways, that's why I am very unsure.   
When using  `SELECT <set quantifier> <select list> <table expression>`, one could say that the aggregation function is implicit since it will filter duplicates.
I think you can leave this as is and see how it goes when doing integration with different tools. 
Maybe add a comment saying that we don't know if this is supposed to mean that DISTINCT and ALL are supported as set quantifier for aggregation function or if they can be used as "standalone" set quantifier?

Reply:
Ok, I left it for now, then. We can remove it if it causes an issue.